chainstate/src/detail/chainstateref/tx_verifier_storage.rs

```diff
@@ -209,4 +220,105 @@ impl<'a, S: BlockchainStorageWrite, O: OrphanBlocks, V: TransactionVerificationS
             }
         }
     }
+
+    fn set_accounting_undo_data(
+        &mut self,
+        tx_source: TransactionSource,
+        undo: &AccountingBlockUndo,
+    ) -> Result<(), TransactionVerifierStorageError> {
+        match tx_source {
+            TransactionSource::Chain(id) => self
+                .db_tx
+                .set_accounting_undo_data(id, undo)
+                .map_err(TransactionVerifierStorageError::from),
+            TransactionSource::Mempool => {
+                panic!("Flushing mempool info into the storage is forbidden")
+            }
```

Comment:
That sounds like a nice idea and it deserves a shot. Let's register it somewhere so that we don't forget. Someone please open an issue.

If we can isolate all mempool vs chain operations at compile time, we'll gain a huge security advantage.

Reply:
done: #633